File: src/main/java/com/ej/management/domain/Task.java
package com.ej.management.domain;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor

public class Task {



    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)

    private Long id;
    private String description;
    private String dateCompleted;
    private String dueDate;
    private Boolean isCompleted;

    @ManyToMany
    List<Person> people;

    @OneToOne
    Category category;


}

File: src/main/front-end/src/app/domain/task.ts
import {Person} from "./person";
import {Category} from "./category";


export class Task {


  public id ?: number;
  public description ?: string;
  public dateCompleted?: string;
  public dueDate ?: string;
  public isCompleted ?: boolean;
  public people ?: Person[];
  public category ?: Category;

  constructor( description: string, dateCompleted: string, dueDate: string, isCompleted: boolean, people: Person[], category: Category) {

    this.description = description;
    this.dateCompleted = dateCompleted;
    this.dueDate = dueDate;
    this.isCompleted = isCompleted;
    this.people = people;
    this.category = category;
  }

}
